.ci/Jenkinsfile

```diff
@@ -24,59 +24,64 @@ pipeline {
     issueCommentTrigger("${obltGitHubComments()}")
   }
   stages {
-    stage('Checkout') {
-      steps {
-        pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
-        deleteDir()
-        gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: true)
-        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
-      }
-    }
-    stage('Install'){
-      steps {
-        withGithubNotify(context: "Install") {
-          withNodeJSEnv() {
-            dir("${BASE_DIR}") {
-              sh(label: 'npm install', script: 'npm install')
-            }
+    stage("Prepare environment") {
+      def environment = docker.build("test-image", "${BASE_DIR}/e2e/Dockerfile")
+      environment.inside {
```

Comment:
Can you confirm this is possible to use? As far as I know, the declarative pipeline syntax is quite restrictive when using groovy specifics, and for such it requires to use the `scripts` closure, normally within the `steps` closure.

Reply:
Yeah, this definitely didn't work. I ended up trying to use the `dockerfile` and `docker` agents to configure build steps. No luck so far, but I have a few more things to attempt. Thanks for your input 🙌